src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -1288,45 +1288,43 @@ void surface_paint(paint_session* session, uint8_t direction, uint16_t height, c
         std::memcpy(session->RightTunnels, backupRightTunnels, sizeof(tunnel_entry) * TUNNEL_MAX_COUNT);
     }
 
-    if (tileElement->AsSurface()->GetWaterHeight() > 0)
+    const uint16_t waterHeight = tileElement->AsSurface()->GetWaterHeight();
+    if ((waterHeight > 0) && !gTrackDesignSaveMode
+        && !((session->ViewFlags & VIEWPORT_FLAG_CLIP_VIEW) && (waterHeight > gClipHeight * COORDS_Z_STEP)))
```

Comment:
That works pretty well, last suggestion would be to make it more readable:
```suggestion
    bool const waterGetsClipped = (session->ViewFlags & VIEWPORT_FLAG_CLIP_VIEW) && waterHeight > gClipHeight * COORDS_Z_STEP;
    if (waterHeight > 0 && !gTrackDesignSaveMode && !waterGetsClipped)
```